## Transcribing Audio

Following The Real Python's [Ultimate Guide To Speech Recognition With Python](https://realpython.com/python-speech-recognition/#picking-a-python-speech-recognition-package), I'm using the SpeechRecognition package to transcribe audio from my microphone to text. Afterwards, I use the TextBlob package to quickly analyze the sentiment.

In [1]:
import speech_recognition as sr
sr.__version__

'3.6.3'

In [2]:
r = sr.Recognizer()

Listing my microphone devices.

In [3]:
mic = sr.Microphone()
sr.Microphone.list_microphone_names()

['Microsoft Sound Mapper - Input',
 'Microphone (Logitech USB Headse',
 'Microsoft Sound Mapper - Output',
 'Speakers (Logitech USB Headset)',
 'Speakers (Realtek High Definiti',
 'Microphone (Realtek HD Audio Mic input)',
 'Stereo Mix (Realtek HD Audio Stereo input)',
 'Line In (Realtek HD Audio Line input)',
 'Speakers (Realtek HD Audio output)',
 'Speakers (Logitech USB Headset)',
 'Microphone (Logitech USB Headset)']

Time to speak something negative into the microphone.

In [4]:
with mic as source:
    audio = r.listen(source)

Using the Google Web Speech API to transcribe my texts. For the moment this will do, though I'm looking into CMU Sphinx as an alternative.

In [5]:
transcript = r.recognize_google(audio)

In [6]:
print(transcript)

time is a flat circle everything I've ever done everything that you will do you will do over and over and over again


Using TextBlob to analyze sentiment. According to [this webpage](https://planspace.org/20150607-textblob_sentiment/), the TextBlob method parses text into tokens, lookups the sentiment, then computes the average sentiment for each token.

In [7]:
from textblob import TextBlob

In [8]:
analyzed_script = TextBlob(transcript)

In [9]:
analyzed_script.sentiment

Sentiment(polarity=-0.025, subjectivity=0.125)

As predicted this was a negative script, since the polarity was -0.025. A textblob can have a polarity between -1 and 1, so this is only slightly negative.

Let's try saying something positive, like this inspirational quote.

In [10]:
with mic as source:
    audio = r.listen(source)

In [11]:
transcript = r.recognize_google(audio)
print(transcript)

our capacity to draw happiness from aesthetic objects or material Goods in fact seems critically dependent on our first satisfying a more important range of emotional or psychological needs among them the need for understanding for love expression and respect


Checking the sentiment:

In [12]:
analyzed_script = TextBlob(transcript)
print(analyzed_script.sentiment)

Sentiment(polarity=0.31666666666666665, subjectivity=0.5759259259259258)


With words like happiness and love in the document, the sentiment has polarity = 0.32. 

In addition there's a measure of subjectivity. The subjectivie ranges from 0 to 1, so this quote is more on the subjective side.